rclpy/src/rclpy/_rclpy.c

```diff
@@ -851,6 +852,99 @@ rclpy_count_subscribers(PyObject * Py_UNUSED(self), PyObject * args)
   return _count_subscribers_publishers(args, "subscribers", rcl_count_subscribers);
 }
 
+typedef rcl_ret_t (* rcl_get_info_by_topic_func_t)(
+  const rcl_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  bool no_mangle,
+  rmw_topic_endpoint_info_array_t * info_array);
+
+static PyObject *
+_get_info_by_topic(
+  PyObject * args, const char * type,
+  rcl_get_info_by_topic_func_t rcl_get_info_by_topic)
+{
+  PyObject * pynode;
+  const char * topic_name;
+  int no_mangle;
+
+  if (!PyArg_ParseTuple(args, "Osp", &pynode, &topic_name, &no_mangle)) {
+    return NULL;
+  }
+
+  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, "rcl_node_t");
+  if (!node) {
+    return NULL;
+  }
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_topic_endpoint_info_array_t info_array = rmw_get_zero_initialized_topic_endpoint_info_array();
+  rcl_ret_t ret = rcl_get_info_by_topic(node, &allocator, topic_name, no_mangle, &info_array);
+  rmw_ret_t fini_ret;
+  if (ret != RCL_RET_OK) {
+    if (ret == RCL_RET_BAD_ALLOC) {
```

Comment:
Consider using:
https://github.com/ros2/rclpy/blob/677de6f23d19c232543e822642f1aa30f6ff4bee/rclpy/src/rclpy/_rclpy.c#L5170
See for example:
https://github.com/ros2/rclpy/blob/677de6f23d19c232543e822642f1aa30f6ff4bee/rclpy/src/rclpy/_rclpy.c#L169-L171

Reply:
Done.